#167190554 Role Based Access Control Functionality

JuwonAbiola opened the pull request:
#### What does this PR do?
- It enables role-based access control

#### Description of Task proposed in this pull request?
- A middleware for authorizing user was created.
- Two columns were added to the User table

#### How should this be manually tested (Quality Assurance)?
- Run npm test to run tests. The middleware can be tested by plugging it into routes.

#### What are the relevant pivotal tracker stories?
- [167190554](https://www.pivotaltracker.com/story/show/167190554)

#### Any background context you want to add (Operations Impact)?
- Run npm run db:rollback to migrate the newly added columns to the database

#### What I have learned working on this feature: 
- I learned how to use ENUM

#### Screenshots: N/A

chialuka commented on server/middlewares/authorizeUser.js:
```diff
@@ -0,0 +1,16 @@
+
+/**
+ * @name authorizeUser
+ * @param {Array} allowedLevel request object
```
Change array to integer, please

JuwonAbiola committed:
feature[167190554]Role based access control functionality
- create middleware that authorize user
- Unit test the middleware
[Delivers #167190554]
